### PR TITLE
Adding MinZip to the archive for Windows.

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds our self-compiled copy of the MinGit executable (built from the flutter/git repo) to the archive when building an archive.

I also tweaked the internal API for prepare_package.dart so that there's a single entry point to build an archive.

### CODE REVIEW

File: dev/bots/prepare_package.dart [hunk 1]
```diff
@@ -2,15 +2,20 @@
...
+import 'dart:typed_data';
 
 import 'package:args/args.dart';
 import 'package:path/path.dart' as path;
+import 'package:http/http.dart' as http;
 
 const String CHROMIUM_REPO =
     'https://chromium.googlesource.com/external/github.com/flutter/flutter';
 const String GITHUB_REPO = 'https://github.com/flutter/flutter.git';
+const String MINGIT_URL = 'https://storage.googleapis.com/flutter_infra/mingit/'
```
Comment (goderbauer) [inline dev/bots/prepare_package.dart]: maybe MINGIT_WIN_URL to emphasize that this is win only?

File: dev/bots/prepare_package.dart [hunk 2]
```diff
@@ -83,15 +99,27 @@ class ArchiveCreator {
     runGit(<String>['remote', 'add', 'origin', GITHUB_REPO]);
   }
 
+  /// Retrieve the MinGit executable from storage and unpack it.
+  Future<Null> installMinGit() async {
```
Comment (goderbauer) [inline dev/bots/prepare_package.dart]: maybe installMinGitForWindows to avoid confusion in the future?

**Review by goderbauer — COMMENTED** [review]
LGTM

@mit-mit can you also review if this bundling is OK as discussed previously?

File: dev/bots/prepare_package.dart (hunk 2 above)
Comment (gspencergoog) [reply dev/bots/prepare_package.dart]: I changed it to installMinGitIfNeeded and if it's not windows, it just returns.

File: dev/bots/prepare_package.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/bots/prepare_package.dart]: Done.

**Comment by mit-mit** [comment]
Didn't review the full script, but I can confirm that bundling mingit.zip and placing it on storage meets the previously discussed requirements, assuming it was build from https://github.com/flutter/git.

**Review by mit-mit — APPROVED** [review]

**Pull request merged**